Plugins/nosMath/Source/Eval.cpp

```diff
@@ -33,12 +33,50 @@ struct EvalNodeContext : NodeContext
 	
 	EvalNodeContext(nosFbNode const* node) : NodeContext(node)
 	{
-		Resolve(node);
+		auto pinCount = node->pins()->size();
+		std::list<nosUUID> pinsToUnorphan;
+		for (auto i = 2; i < pinCount; i++)
+		{
+			auto pin = node->pins()->Get(i);
+			if (pin->show_as() == fb::ShowAs::INPUT_PIN
+				// Flag pins
+				&& pin->name()->string_view() != "Show_Expression")
+			{
+				auto uniqueName = pin->name()->str();
+				auto displayName = pin->display_name() ? pin->display_name()->str() : uniqueName;
+				Variables[nos::Name(uniqueName)] = 0.0;
+				Inputs.push_back(*pin->id());
+				if (auto orphanState = pin->orphan_state()) {
+					if (orphanState->type() == fb::PinOrphanStateType::ORPHAN)
+						pinsToUnorphan.push_back(*pin->id());
+				}
+			}
+		}
+		Compile();
+		for (auto const& pinId : pinsToUnorphan)
+			SetPinOrphanState(pinId, fb::PinOrphanStateType::ACTIVE);
+	}
+
+	void OnPartialNodeUpdated(nosNodeUpdate const* update) override
+	{
+		if (update->Type != NOS_NODE_UPDATE_PIN_DELETED)
+			return;
+		// Since pin is deleted from the NodeContext's pins map, we can't get its name from its id.
```

Comment:
We thought changing `Variables` member to a pin ID map instead of unique name, could make this better.